### ISSUE TITLE
Automate Online Documentation

### ISSUE THREAD

**Issue by jan-abel-inwt** [Pandora-IsoMemo/MapR#18]
When https://github.com/Pandora-IsoMemo/MapR/pull/17 is merged into the main branch:

- [x] Set Up Github Pages 
- [x] Automatically build pkgdown documentation on main branch via Actions
- [x] We need to add the link to the HTML documentation to the readMe, e.g. just put the link of the HTML vignette under the word "vignette" above. (as soon it is available online.)
- [x] to improve future development, i also will add an r-cmd check via gh-actions

**Comment by jan-abel-inwt** [comment]
I am using this [workflow example](https://github.com/r-lib/actions/blob/v2/examples/pkgdown.yaml) via the `usethis` package to set up the online documentation.

0. Add a PAT to work with `usethis` pkg

```R
usethis::create_github_token()
gitcreds::gitcreds_set()
```

1. I added the new branch `gh-pages` that we are only using for publishing the documentation 

```R
> usethis::use_github_pages()
✔ Initializing empty, orphan 'gh-pages' branch in GitHub repo 'Pandora-IsoMemo/MapR'
✔ GitHub Pages is publishing from:
• URL: 'https://pandora-isomemo.github.io/MapR/'
• Branch: 'gh-pages'
• Path: '/'
```

2. The next step implements the GitHub setup needed to automatically publish the pkgdown site to GitHub pages

```R
> usethis::use_pkgdown_github_pages()
✔ Adding '^_pkgdown\\.yml$', '^docs$', '^pkgdown$' to '.Rbuildignore'
✔ Adding 'docs' to '.gitignore'
✔ Writing '_pkgdown.yml'
• Modify '_pkgdown.yml'
✔ GitHub Pages is publishing from:
• URL: 'https://pandora-isomemo.github.io/MapR/'
• Branch: 'gh-pages'
• Path: '/'
✔ Adding '^\\.github$' to '.Rbuildignore'
✔ Adding '*.html' to '.github/.gitignore'
✔ Saving 'r-lib/actions/examples/pkgdown.yaml@v2' to '.github/workflows/pkgdown.yaml'
• Learn more at <https://github.com/r-lib/actions/blob/v2/examples/README.md>.
✔ Recording 'https://pandora-isomemo.github.io/MapR/' as site's url in '_pkgdown.yml'
✔ Adding 'https://pandora-isomemo.github.io/MapR/' to URL
✔ Setting 'https://pandora-isomemo.github.io/MapR/' as homepage of GitHub repo 'Pandora-IsoMemo/MapR'
```

**Comment by jan-abel-inwt** [comment]
This task wasn't easy, and it took quite some time to figure out. I encountered difficulties with the package manager 'pak' in the GitHub Action workflow, as it couldn't find 'datatools' and 'rgpt3'. Despite trying various solutions, none of them worked.

It turns out that the issue arises from using a remote package ('rgpt3') in a CRAN-like repository (our 'drat' repo)

This  doesn't work, because `pak` relies on the metadata in the [Packages file](https://github.com/Pandora-IsoMemo/drat/blob/main/docs/src/contrib/PACKAGES) of the `drat` repository and this file don't contain the remotes information.

This is likely why we had to manually load 'rgpt3' in every package connected to 'DataTools'.

The proposed solution is to [integrate the rgpt3 functions into `DataTools](https://github.com/Pandora-IsoMemo/data-tools/issues/68). This could potentially make things work more smoothly and simplify our dependency management.
